### PR TITLE
Fix to BUG GH11698. added default value of mask

### LINKED ISSUE #11698: pandas.DataFrame.replace raises UnboundLocalError on mixed data types
This is in 0.17.1. Suppose we have a frame with mixed data types. Suppose also we want to replace a string value with something else. If the frame has only string columns, or maybe string and int columns, this works fine. However, if the frame has string and datetime columns, this raises an exception. Example:

```
>>> pandas.DataFrame([('-', pandas.to_datetime('20150101')), ('a', pandas.to_datetime('20150102')), ('b', pandas.to_datetime('20150103'))], columns=['a', 'b']).replace('-', numpy.nan)
---------------------------------------------------------------------------
UnboundLocalError                         Traceback (most recent call last)
<ipython-input-11-bce7dc78fc44> in <module>()
----> 1 pandas.DataFrame([('-', pandas.to_datetime('20150101')), ('a', pandas.to_datetime('20150102')), ('b', pandas.to_datetime('20150103'))], columns=['a', 'b']).replace('-', numpy.nan)

/.../pandas/core/generic.pyc in replace(self, to_replace, value, inplace, limit, regex, method, axis)
   3108                 elif not com.is_list_like(value):  # NA -> 0
   3109                     new_data = self._data.replace(to_replace=to_replace, value=value,
-> 3110                                                   inplace=inplace, regex=regex)
   3111                 else:
   3112                     msg = ('Invalid "to_replace" type: '

/.../pandas/core/internals.pyc in replace(self, **kwargs)
   2868
   2869     def replace(self, **kwargs):
-> 2870         return self.apply('replace', **kwargs)
   2871
   2872     def replace_list(self, src_list, dest_list, inplace=False, regex=False, mgr=None):

/.../pandas/core/internals.pyc in apply(self, f, axes, filter, do_integrity_check, consolidate, **kwargs)
   2821
   2822             kwargs['mgr'] = self
-> 2823             applied = getattr(b, f)(**kwargs)
   2824             result_blocks = _extend_blocks(applied, result_blocks)
   2825

/.../pandas/core/internals.pyc in replace(self, to_replace, value, inplace, filter, regex, convert, mgr)
    605
    606             # we can't process the value, but nothing to do
--> 607             if not mask.any():
    608                 return self if inplace else self.copy()
    609

UnboundLocalError: local variable 'mask' referenced before assignment
```

And the requisite show_versions:

```
INSTALLED VERSIONS
------------------
commit: None
python: 2.7.10.final.0
python-bits: 64
OS: Linux
OS-release: 2.6.18-348.2.1.el5
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: C
LANG: en_US.UTF-8

pandas: 0.17.1
nose: 1.3.7
pip: 7.1.2
setuptools: 18.4
Cython: 0.23.4
numpy: 1.10.1
scipy: 0.16.0
statsmodels: 0.6.1
IPython: 4.0.0
sphinx: 1.3.1
patsy: 0.4.0
dateutil: 2.4.2
pytz: 2015.7
blosc: None
bottleneck: 1.0.0
tables: 3.1.0
numexpr: 2.4.4
matplotlib: 1.5.0
openpyxl: 2.2.6
xlrd: 0.9.4
xlwt: 1.0.0
xlsxwriter: 0.7.7
lxml: 3.4.4
bs4: 4.3.2
html5lib: 0.999
httplib2: None
apiclient: None
sqlalchemy: 1.0.9
pymysql: None
psycopg2: None
Jinja2: None
```

### PR DESCRIPTION
**Pull request by varunentropy**
closes #11698

Please review.

### CODE REVIEW

File: pandas/tests/test_frame.py [hunk 1]
```diff
@@ -9140,6 +9140,13 @@ def test_replace_inplace(self):
         tsframe.replace([nan], [0], inplace=True)
         assert_frame_equal(tsframe, self.tsframe.fillna(0))
 
+        # GH 11698
+        # test for mixed data types.
+        df = pd.DataFrame([('-', pd.to_datetime('20150101')), ('a', pd.to_datetime('20150102'))])
+        df.replace('-', np.nan, inplace=True)
```
Comment (jreback) [inline pandas/tests/test_frame.py]: use ``result = df.replace(....)``

never use ``inplace`` in a test, unless you are explicity testing something that involves it explicity. this is not pandonic at all.

**Comment by jreback** [comment]
minor change, ping when green.

**Comment by varun-kr** [comment]
modified test case. 
This branch has some conflicts . Please check.

**Comment by jreback** [comment]
u need to fix the conflict

**Comment by varun-kr** [comment]
@jreback  There is a conflict in whatsnew file. When I am adding the bug fix message and trying to merge with the latest code , it always shows me conflicts. Please help.

**Comment by jreback** [comment]
lots of help on this: http://githowto.com/resolving_conflicts

you basically edit the conflicted file, picking the correct parts out. then add the files, rebase, and push

**Comment by jreback** [comment]
merged via https://github.com/pydata/pandas/commit/8ec8487a4a8362ac94630388f217f273b3ad1525

thanks!

**Comment by varun-kr** [comment]
Thanks

**Pull request closed without merging**